docs/grammar.txt

```diff
@@ -100,6 +105,8 @@ Identifier = [a-zA-Z_] [a-zA-Z_0-9]*
 HexNumber = '0x' [0-9a-fA-F]+
 DecimalNumber = [0-9]+
 
+ElementaryTypeNameExpression = ElementaryTypeName '(' expression ')'
```

Comment:
Thanks for clearing this up for me! I discovered it last night while trying to improve my understanding of the parser code. Indeed, Solidity accepts and compiles the following source code without raising any errors:
````
contract test {
  function() { uint; }
}
````
And it can apparently handle this too:
````
var a = uint;
a(1);
````
But I didn't see this behavior documented anywhere.